File: 2125. Number of Laser Beams in a Bank/2125-Number-of-Laser-Beams-in-a-Bank.cs
public class Solution {
    public int NumberOfBeams(string[] bank) {
        int prev = 0, curr = 0, sum = 0;
        for(var i=0; i<bank.Length; i++){
            curr = bank[i].Count(c => c == '1');
            sum += curr*prev;
            if(curr != 0)
                prev = curr;
        }
        return sum;
    }
}

File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int prev = 0, curr = 0, sum = 0;
        for(var bnk : bank){
            curr = 0;
            for(var ch : bnk.toCharArray())
                if(ch == '1') curr++;
            sum += curr * prev;
            if(curr != 0)
                prev = curr;
        }
        return sum;
    }
}